src/flake8_tidy_imports/__init__.py

```diff
@@ -92,11 +102,18 @@ def parse_options(cls, options: Any) -> None:
 
         cls.ban_relative_imports = options.ban_relative_imports
 
+        parsed_import_statements = cls.parse_import_idioms(options.import_idioms)
+        for statement in parsed_import_statements:
+            idioms = cls.compose_idioms(statement)
+            for idiom in idioms:
+                cls.import_idioms.update(idiom)
```

Comment:
I don't like this two-step parsing procedure with regexes then the AST. We can use solely `ast.parse`, which will allow users to write their idioms with whichever syntax they like.

Also the parse method should create the final `import_idioms` data structure, rather than scattering that logic between the parsing function and `parse_options`